### ISSUE TITLE
Booting Image with QEMU fails

### ISSUE THREAD

**Issue by SteffenDE** [nerves-project/nerves_system_x86_64#129]
### Environment

* Elixir version (`elixir -v`):
```
elixir -v
Erlang/OTP 24 [erts-12.0.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Elixir 1.12.2 (compiled with Erlang/OTP 24)
```
* Nerves environment: (`mix nerves.env --info`)
```
MIX_TARGET=x86_64 mix nerves.env --info
==> nerves
==> minimal
|nerves_bootstrap| Environment Package List

  Pkg:         nerves_system_br
  Vsn:         1.16.1
  Type:        system_platform
  BuildRunner: {nil, []}

  Pkg:         nerves_system_x86_64
  Vsn:         1.16.1
  Type:        system
  BuildRunner: {Nerves.Artifact.BuildRunners.Local, []}

  Pkg:         nerves_toolchain_ctng
  Vsn:         1.8.4
  Type:        toolchain_platform
  BuildRunner: {nil, []}

  Pkg:         nerves_toolchain_x86_64_nerves_linux_musl
  Vsn:         1.4.3
  Type:        toolchain
  BuildRunner: {Nerves.Artifact.BuildRunners.Local, []}

|nerves_bootstrap| Loadpaths Start


Nerves environment
  MIX_TARGET:   x86_64
  MIX_ENV:      dev

|nerves_bootstrap| Environment Variable List
  target:     x86_64
  toolchain:  /root/.nerves/artifacts/nerves_toolchain_x86_64_nerves_linux_musl-linux_x86_64-1.4.3
  system:     /root/.nerves/artifacts/nerves_system_x86_64-portable-1.16.1
  app:        .

|nerves_bootstrap| Loadpaths End
```

### Current behavior

Hi there,
I'm currently trying to boot a nerves image using QEMU for local development. I used the nerves_examples repo with the minimal example:
```bash
$ git clone https://github.com/nerves-project/nerves_examples.git
$ cd nerves_examples/minimal/
$ MIX_TARGET=x86_64 mix deps.get
$ MIX_TARGET=x86_64 mix firmware.image
$ cp minimal.img /var/lib/libvirt/images/nerves.img
$ virt-install --memory 1024 --import --disk /var/lib/libvirt/images/nerves.img
```
The resulting machine starts to boot and pauses at the "GRUB loading.." screen.

![image](https://user-images.githubusercontent.com/4116351/126649445-b0a38501-46b5-42ba-8c1a-a5ed0b2acaa4.png)

The goal is to boot a project on an Intel NUC (which actually needs an EFI bootloader, therefore I adapted this repo based on the changes from https://github.com/kbredemeier/nerves_system_x86_64_uefi), but there the boot process also does not go any further than the GRUB loading screen. I think that maybe this is related, therefore I want to get a basic image to boot with QEMU first.

I can actually get the image to boot using another disk with an existing GRUB installation, specifying the correct root PARTUUID, therefore I think that there's something wrong with GRUB:
![image](https://user-images.githubusercontent.com/4116351/126651706-7bbcacf8-be13-47cc-96d6-402a46d6c9f7.png)
Note that hd0 is in this case the other non-Nerves image that contains a bootable Linux image with GRUB.
![image](https://user-images.githubusercontent.com/4116351/126652082-3049c43e-02ab-44ce-83c6-c0869be16935.png)

### Expected behavior

The machine boots.

**Comment by SteffenDE** [comment]
![image](https://user-images.githubusercontent.com/4116351/129476198-a6622b4a-5ef9-48c9-b155-f491e5f12e83.png)

Latest working version: nerves_system_x86_64 release 1.15.0, breaks with 1.15.1.

Tested with the minimal example, manually specifying the dependency:
```elixir
      # Dependencies for specific targets
      {:nerves_system_x86_64, "== 1.15.0", runtime: false, targets: :x86_64}
```

**Comment by fhunleth** [comment]
Thanks for the update. 

I fixed the easy issue: https://github.com/nerves-project/nerves_examples/pull/215
I'm not sure yet on why 1.15.1 breaks.

**Comment by fhunleth** [comment]
That's really weird that 1.15.0 works and 1.15.1 breaks. The only two changes are a security release bump for Buildroot and  a minor Erlang/OTP bump. It would be interesting to know how it breaks. 

Also, since I think the most changes were in the Buildroot change, here's the Buildroot announcement: http://lists.busybox.net/pipermail/buildroot/2021-April/307970.html.

**Comment by SteffenDE** [comment]
I guess it's an issue with one of the grub patches added in 2021.02.1 (https://github.com/buildroot/buildroot/compare/2021.02...2021.02.1), but I'm not sure how to track this down further...

**Comment by fhunleth** [comment]
Do you need to use `grub`? We switched to just using the UEFI bootloader at my work. I don't directly work on that project, but I can check if there's anything new that can be open-sourced from it. We don't use QEMU, though. I think that https://github.com/fhunleth/nerves_system_up_board is closest to what we're using internally.

**Comment by SteffenDE** [comment]
Alright, I will try the `nerves_system_up_board` - actually I target an Intel NUC, I only wanted to use QEMU for testing.
Still, I think that the `nerves_system_x86_64` system should be able to boot on most x86 based PCs, which currently does not seem to be the case... Do you know someone that uses the current version successfully?

**Comment by SteffenDE** [comment]
Okay that's strange. I just cloned the latest version of the nerves-examples repo, as I realised that there still exists the mix task for generating a QEMU script:
```bash
$ git clone https://github.com/nerves-project/nerves_examples.git
$ cd nerves_examples/minimal/
$ MIX_TARGET=x86_64 mix deps.get
$ MIX_TARGET=x86_64 mix firmware.image
$ MIX_TARGET=x86_64 mix nerves.gen.qemu_script
$ ./run-qemu.sh
```
With the latest 1.16.2 release of `nerves_system_x86_64` booting with QEMU started to work again! So it seems that this is fixed with the latest nerves_system_br using Buildroot 2021.05.1 😃

**Comment by SteffenDE** [comment]
Probably this commit fixed it: https://github.com/buildroot/buildroot/commit/ef2a07f8c6060eef20c8b95bf694db388f6d8f06

**Comment by fhunleth** [comment]
I'm glad that you found that Buildroot commit. That sounds like a painful bug.

In answer to your question, I don't know anyone using `nerves_system_x86_64`. Historically, it was a starter system since the x86 devices people were using at the time were different enough that one system wasn't possible. Then we tried using it for qemu and abandoned that effort since Raspberry Pi Zero's were easier to use and $5-$10 each (in the US).

I agree with your statement about `nerves_system_x86_64` booting on most PCs. Everything x86_64 that I've used in the past couple years has been EFI, and NUC projects seem really common. I think it's a matter of finding someone in the Nerves community to spearhead a good Nerves system for x86_64. Sadly, I don't have time and my x86_64 projects are moving to ARM (for non-technical reasons). It would be great to replace `nerves_system_x86_64`with something that could actually be used out-of-the-box.
